src/app/core/basic-datatypes/entity/entity.datatype.spec.ts

```diff
@@ -23,50 +23,76 @@ import { EntitySchemaField } from "../../entity/schema/entity-schema-field";
 import { TestEntity } from "../../../utils/test-utils/TestEntity";
 
 describe("Schema data type: entity", () => {
-  testDatatype(new EntityDatatype(null, null), "1", "1", "User");
+  testDatatype(new EntityDatatype(null as any, null as any), "1", "1", "User");
 
-  it("should map to the referenced entity", async () => {
-    const c1 = TestEntity.create("first");
-    const c2 = new TestEntity();
-    c2.other = "123";
-    const entityMapper = mockEntityMapper([c1, c2]);
-    const dataType = new EntityDatatype(entityMapper, null);
-    const schema = TestEntity.schema.get("ref");
+  describe("importMapFunction", () => {
+    let entityMapper: ReturnType<typeof mockEntityMapper>;
+    let dataType: EntityDatatype;
+    let schema: EntitySchemaField;
 
-    await expectAsync(
-      dataType.importMapFunction("first", schema, "name"),
-    ).toBeResolvedTo(c1.getId());
-    await expectAsync(
-      dataType.importMapFunction("123", schema, "other"),
-    ).toBeResolvedTo(c2.getId());
-    await expectAsync(
-      dataType.importMapFunction("345", schema, "other"),
-    ).toBeResolvedTo(undefined);
+    let c1: TestEntity;
+    let c2: TestEntity;
+
+    beforeEach(() => {
+      c1 = TestEntity.create("first");
+      c2 = new TestEntity();
+      c2.other = "123"; // Ensure other is a string
+      entityMapper = mockEntityMapper([c1, c2]);
+      dataType = new EntityDatatype(entityMapper, null as any);
+      schema = TestEntity.schema.get("ref") as EntitySchemaField;
+    });
+
+    it("should map to the referenced entity by name", async () => {
+      await expectAsync(dataType.importMapFunction("first", schema, "name"))
+        .toBeResolvedTo(c1.getId());
+    });
+
+    it("should map to the referenced entity by other field", async () => {
+      await expectAsync(dataType.importMapFunction("123", schema, "other"))
+        .toBeResolvedTo(c2.getId());
+    });
+
+    it("should return undefined when no matching entity is found", async () => {
+      await expectAsync(dataType.importMapFunction("345", schema, "other"))
+        .toBeResolvedTo(undefined);
+    });
+
+    it("should handle numeric-string mismatches correctly", async () => {
+      const c3 = new TestEntity();
+      c3.other = "456"; // Ensure it's a string
+      entityMapper = mockEntityMapper([c3]);
+      dataType = new EntityDatatype(entityMapper, null as any);
+      
+      await expectAsync(dataType.importMapFunction("456", schema, "other"))
+        .toBeResolvedTo(c3.getId());
```

Comment:
I think the following test case should cover the requirements of this issue:

```suggestion
      // "simple" case: imported value is string already
      await expectAsync(dataType.importMapFunction("456", schema, "other"))
        .toBeResolvedTo(c3.getId());
      // "advanced" case: imported value is number but should match also
      await expectAsync(dataType.importMapFunction(456, schema, "other"))
        .toBeResolvedTo(c3.getId());
```